checks.d/riakcs.py

```diff
@@ -0,0 +1,97 @@
+# stdlib
+from collections import defaultdict
+
+# project
+from checks import AgentCheck
+
+# 3rd party
+import simplejson as json
+from boto.s3.connection import S3Connection
+
+
+def multidict(ordered_pairs):
+    """Convert duplicate keys values to lists."""
+    # read all values into lists
+    d = defaultdict(list)
+    for k, v in ordered_pairs:
+        d[k].append(v)
+    # unpack lists that have only 1 item
+    for k, v in d.items():
+        if len(v) == 1:
+            d[k] = v[0]
+    return dict(d)
+
+
+
+
+class RiakCs(AgentCheck):
+
+    STATS_BUCKET = 'riak-cs'
+    STATS_KEY = 'stats'
+
+    def check(self, instance):
+        s3, aggregation_key, tags = self._connect(instance)
+
+        stats = self._get_stats(s3, aggregation_key)
+        self.process_stats(stats, tags)
+
+    def process_stats(self, stats, tags):
+        if not stats:
+            raise Exception("No stats were collected")
+
+        legends = dict([(len(k), k) for k in stats["legend"]])
+        del stats["legend"]
+        for key, values in stats.iteritems():
+            legend = legends[len(values)]
+            for i, value in enumerate(values):
+                metric_name = "riakcs.{0}.{1}".format(key, legend[i])
+                self.gauge(metric_name, value, tags=tags)
+
+
+    def _connect(self, instance):
+        for e in ("access_id", "access_secret"):
+            if e not in instance:
+                raise Exception("{0} parameter is required.".format(e))
+
+        s3_settings={
+            "aws_access_key_id": instance.get('access_id', None),
+            "aws_secret_access_key": instance.get('access_secret', None),
+            "proxy": instance.get('host','localhost'),
+            "proxy_port": instance.get('port', 8080),
+            "is_secure": instance.get('is_secure', None)
+        }
+
+        if instance.get('s3_root'):
+            s3_settings['host'] = instance['s3_root']
+
+        aggregation_key = s3_settings['proxy'] + ":" + str(s3_settings['proxy_port'])
+
+        try:
+            s3 = S3Connection(**s3_settings)
+        except Exception, e:
+            self.log.error("Error connecting to " + aggregation_key, e)
```

Comment:
You probably don't want to do that, this syntax is reserved to string placeholders, an exception will probably raise a type error.